colossalai/initialize.py

```diff
@@ -324,7 +338,7 @@ def initialize(model: nn.Module,
         # 1. if optimizer is ZERO, then use zero grad handler
         # 2. if dp size is larger than 1 and pipeline is not used, use pytorch ddp
         # 3. if using pipeline and dp size larger than 1, use data parallel grad handler
-        if isinstance(optimizer, ShardedOptimizer):
+        if isinstance(optimizer, ShardedOptimizerV2):
             gradient_handler_cfg = [dict(type='ZeROGradientHandler')]
```

Comment:
Do we still need `ZeROGradientHandler` now?

Reply:
I don't think so.

Reply:
Keep it now, remove if necessary.

Reply:
Keeping it will lead to exception as this gradient handler requires the optimizer to have a `sync_grad` method.

Reply:
OK, I have already added such a method in ShardedOptimV2